code/modules/library/barcode_scanner.dm

```diff
@@ -0,0 +1,41 @@
+/*
+ * Barcode Scanner
+ */
+/obj/item/barcodescanner
+	name = "barcode scanner"
+	icon = 'icons/obj/library.dmi'
+	icon_state ="scanner"
+	desc = "A fabulous tool if you need to scan a barcode."
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_TINY
+	/// A weakref to our associated computer - Modes 1 to 3 use this
+	var/datum/weakref/computer_ref
+	/// Currently scanned book
+	var/datum/book_info/book_data
+	/// 0 - Scan only, 1 - Scan and Set Buffer, 2 - Scan and Attempt to Check In, 3 - Scan and Attempt to Add to Inventory
```

Comment:
make these use defines

Reply:
I initially just split the large ass file into a bunch of other smaller files with no changes made.  Barcode scanners and the other files that got moved have nothing to do with my PR when I just wanted the book code to be in it's own file.

I made the changes in 6485908478c30516514bfe896df7ab726a703611 but I'm going to start being more atomic with my PRs in the future.